### ISSUE TITLE
upper method in tt files gives an empty string

### ISSUE THREAD

**Issue by StephaneGerardVUB** [quattor/CAF#88]
Hi,
I'm facing this issue when I try to define a vnet with ncm-opennebula on a SL6.6 machine. Here is a code that isolates the problem :
```
#!/usr/bin/perl
use lib '/usr/lib/perl';
use CAF::TextRender;
my $module = 'vnet';
my %vnet = ('bridge' , 'br0', 'dns', '193.58.172.5', 'gateway', '192.168.10.200', 'name', 'Private', 'network_mask', '255.255.0.0', 'type', 'FIXED');
my $contents = \%vnet;
my $trd = CAF::TextRender->new($module, {'vnet' => $contents}, relpath => 'opennebula', log => $self);
print "$trd";
```
The ouput of this code is the following :
```
 = "br0"
 = "193.58.172.5"
 = "192.168.10.200"
 = "Private"
 = "255.255.0.0"
 = "FIXED"
QUATTOR = 1
```
I guess this is due to "upper" methode in vnet.tt :
```
[% FOR pair IN vnet.pairs -%]
[%    pair.key.upper %] = "[% pair.value %]"
[% END -%]
QUATTOR = 1
```
Could this be due to bug in perl-Template-Toolkit-2.22-5.el6.x86_64 ?
Cheers,
Stéphane

**Comment by stdweird** [comment]
@StephaneGerardVUB yes, `.upper` doesn't work with older perl-Template-Toolkit. we should use `pair.key FILTER upper` for that (see also https://github.com/stdweird/configuration-modules-core/commit/f837f952d189a8fe4f83aa748dd4b43aa7b36e80)

@jrha as soon as we will run the unittests on EL6 for the release, this would have popped up :wink: 

@alvarosimon can you look for this (and we need to run the tests under EL6 asap)

**Comment by alvarosimon** [comment]
@stdweird, @StephaneGerardVUB yes I will test this in a EL6 as soon as possible

**Comment by alvarosimon** [comment]
@stdweird opennebula AII is not affected by this, I will modify the current opennebula component PR to fix this asap

**Comment by alvarosimon** [comment]
Updated opennebula component PR: https://github.com/quattor/configuration-modules-core/pull/466 to fix this issue

**Comment by jrha** [comment]
Fixed by quattor/configuration-modules-core#466.
